apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -59,14 +59,16 @@ internal SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<Sub
         GaxPreconditions.CheckNotNull(settings, nameof(settings));
         settings.Validate();
         // These values are validated in Settings.Validate() above, so no need to re-validate here.
-        _modifyDeadlineSeconds = (int)((settings.AckDeadline ?? DefaultAckDeadline).TotalSeconds);
-        var autoExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds) - (settings.AckExtensionWindow ?? DefaultAckExtensionWindow);
-        var autoExtendIntervalForExactlyOnceDelivery = settings.AckExtensionWindow ?? MinimumAckExtensionWindowForExactlyOnceDelivery;
-        // For exactly once delivery subscription, minimum ack extension window value should be the default value of 60 seconds or the user provided value.
+        _ackExtensionWindow = settings.AckExtensionWindow ?? DefaultAckExtensionWindow;
+        _modifyDeadlineSeconds = (int) ((settings.AckDeadline ?? DefaultAckDeadline).TotalSeconds);
+        // For exactly-once delivery subscriptions, AckDeadline value should be the default value of 60 seconds or the user provided value. 
+        // Due to both DefaultAckDeadline and MinimumAckDeadlineExtensionForExactlyOnceDelivery being 60 second,
+        // AckDeadline for regular and eod subs have same value currently. But this is just coincidence. It is better to keep them seperate.
```

Comment:
I'm not sure this is right. I think that the  effective AckDeadline for eod cannot be less than 60 seconds even if the user specifies so. `MinimumAckDeadlineExtensionForExactlyOnceDelivery` is not a default value, it's a constraint on a value. So for me it would go something like:

```
var maybeInvalid = settings.Ackdealine ?? DefaultAckDeadline;
_modifyDeadlineSeconds... = maybeInvalid < MinimumAckDeadline... ? MinimumAckDeadline : maybeInvalid;
```

Reply:
The requirement for eod is that the `MinAckDeadline` should be either:

- A user specified value in range (10, 600), or
- The default minimum Ack Deadline extension of 60 seconds if user hasn't specified the MinAckDeadline.

Now, this requirement comes in the document considering the libraries that takes the MinimumAckDeadline, MaximumAckDeadline, TotalMaxAckDeadline for lease management. They don't take AckDeadline as input from the user.

We do have default AckDeadline as 60 seconds which is what we had used earlier. But keeping in view of above requirements, I changed to use MinimumAckDeadline instead of the DefaultAckDeadline if the user has not provided AckDeadline.